Purchases/Public/RCPurchases.m

```diff
@@ -652,84 +652,75 @@ - (void)updatePurchaserInfoCache:(RCReceivePurchaserInfoBlock _Nullable)completi
-                                           [productIdentifiers addObject:offering.activeProductIdentifier];
-                                       }];
-
-                                       [self productsWithIdentifiers:productIdentifiers.allObjects completionBlock:^(NSArray<SKProduct *> * _Nonnull products) {
-                                           NSMutableDictionary *productsById = [NSMutableDictionary new];
-                                           for (SKProduct *p in products) {
-                                               productsById[p.productIdentifier] = p;
-                                           }
-                                           
-                                           NSMutableArray *missingProducts = [NSMutableArray new];
-
-                                           [self performOnEachOfferingInEntitlements:entitlements block:^(RCOffering *offering) {
-                                               SKProduct *product = productsById[offering.activeProductIdentifier];
-                                               
-                                               
-                                               if (product == nil) {
-                                                   [missingProducts addObject:offering.activeProductIdentifier];
-                                               }
-                                               
-                                               offering.activeProduct = product;
-                                           }];
-                                           
-                                           if (missingProducts.count > 0) {
-                                               RCLog(@"Could not find SKProduct for %@", missingProducts);
-                                               RCLog(@"Ensure your products are correctly configured in App Store Connect");
-                                               RCLog(@"See https://www.revenuecat.com/2018/10/11/configuring-in-app-products-is-hard");
-                                           }
-
-                                           if (entitlements != nil) {
-                                               self.cachedEntitlements = entitlements;
-                                           }
-                                           
-                                           CALL_AND_DISPATCH_IF_SET(completion, entitlements, nil);
-                                       }];
-    }];
+                                        NSMutableDictionary *productsById = [NSMutableDictionary new];
+                                        for (SKProduct *p in products) {
+                                            productsById[p.productIdentifier] = p;
+                                        }
+                                        RCOfferings *offerings = [RCOfferingsFactory createOfferingsWithProducts:productsById data:data];
```

Comment:
When using a factory, usually it's best practice to inject the factory as a dependency.